File: saude_ruby/lib/saude_ruby/imc.rb
# lib/saude_ruby/imc.rb
module SaudeRuby
  class IMC
    def self.calcular(peso, altura)
      raise "Altura deve ser maior que zero" if altura <= 0
      imc = peso / (altura ** 2)
      format("%.2f", imc).to_f
    end

    def self.classificacao(imc)
      case imc
      when 0..18.4 then "Abaixo do peso"
      when 18.5..24.9 then "Peso normal"
      when 25..29.9 then "Sobrepeso"
      when 30..34.9 then "Obesidade Grau 1"
      when 35..39.9 then "Obesidade Grau 2"
      else "Obesidade Grau 3 (mórbida)"
      end
    end
  end
end

File: saude_javascript/lib/imc.js
class IMC {
    static calcular(peso, altura) {
      if (altura <= 0) {
        throw new Error('Altura deve ser maior que zero');
      }
      const imc = peso / (altura ** 2);
      return parseFloat(imc.toFixed(2));
    }
  
    static classificacao(imc) {
      if (imc < 18.5) {
        return 'Abaixo do peso';
      } else if (imc >= 18.5 && imc <= 24.9) {
        return 'Peso normal';
      } else if (imc >= 25 && imc <= 29.9) {
        return 'Sobrepeso';
      } else if (imc >= 30 && imc <= 34.9) {
        return 'Obesidade Grau 1';
      } else if (imc >= 35 && imc <= 39.9) {
        return 'Obesidade Grau 2';
      } else {
        return 'Obesidade Grau 3 (mórbida)';
      }
    }
  }
  
  module.exports = IMC;
  